aodndata/gsla/handler.py

```diff
@@ -0,0 +1,158 @@
+        return "NRT00"
+    elif "_GSLA_FV02_DM00_" in os.path.basename(filepath):
+        return "DM00"
+    try:
+        get_fields_from_filename(filepath, GSLA_REGEX_YEARLY)
+        return "DM00/yearfiles"
+    except AttributeError:
+        raise InvalidFileNameError(
+            "file name: \"{filename}\" not matching regex to deduce dest_path".format(
+                filename=os.path.basename(filepath)))
+
+
+class GslaHandler(HandlerBase):
+    def __init__(self, *args, **kwargs):
+        super(GslaHandler, self).__init__(*args, **kwargs)
+        self.allowed_extensions = ['.gz']
+
+    def preprocess(self):
+        # add nc_gz file to collection (not by default)
+        self.file_collection.add(self.input_file_object)
+        nc_gz_collection = self.file_collection.filter_by_attribute_id('file_type', FileType.GZIP)
+
+        # GSLA files are gzipped, so gunzip them before checking them
+        if self.file_type is FileType.GZIP:
+            # if uploaded file is GZIP check that GZIP contains a NetCDF
+            netcdf = self.file_collection.filter_by_attribute_id('file_type', FileType.NETCDF)
+            if len(netcdf) != 1:
+                raise InvalidInputFileError(
+                    "Expecting one netCDF file in GZIP archive '{gzip}'".format(gzip=os.path.basename(self.input_file))
+                )
+
+            nc = netcdf[0]
+            result_previous_version_creation_date = self.get_previous_version_creation_date(nc.name)
+
+            push_new_file = True  # by default we push to the storage the file landed in the pipeline
+            remove_previous_version = False
+
+            # check creation date of file already on storage
+            if result_previous_version_creation_date:
+                new_file_creation_date = get_creation_date(nc.name)
+                if result_previous_version_creation_date > new_file_creation_date:
+                    push_new_file = False
+                else:
+                    remove_previous_version = True
+                    previous_file_object = self.get_previous_version_object(nc.name)
+
+            if push_new_file:
+                # index un-gzipped file, but push gzipped file to S3
+                nc.publish_type = PipelineFilePublishType.HARVEST_ONLY
+                nc_gz_collection.set_publish_types(PipelineFilePublishType.UPLOAD_ONLY)  # to set to all files in the collection
+
+            else:
+                nc.publish_type = PipelineFilePublishType.NO_ACTION
+                nc_gz_collection.publish_type = PipelineFilePublishType.NO_ACTION
+
+            # deletion of the previous file
+            if remove_previous_version:
+                previous_file_name = os.path.basename(previous_file_object.keys()[0])
+                file_to_delete = PipelineFile(previous_file_name,
+                                              is_deletion=True)
+                self.file_collection.add(file_to_delete)
```

Comment:
but in that case how do I run queries such as
```python
        netcdf_file_gz_collection = self.file_collection.filter_by_attribute_id('file_type', FileType.GZIP)
```

Reply:
Sorry I might not have been clear. I just meant reorder these 3 lines so that the .add() call is last, e.g.

```python
self.file_collection.add(file_to_delete)
file_to_delete.dest_path = self.dest_path(previous_file_name)
file_to_delete.publish_type = PipelineFilePublishType.DELETE_ONLY
```

becomes

```python
file_to_delete.dest_path = self.dest_path(previous_file_name)
file_to_delete.publish_type = PipelineFilePublishType.DELETE_ONLY
self.file_collection.add(file_to_delete)
```